Source/WebKit/WebProcess/Extensions/Bindings/Cocoa/JSWebExtensionWrapperCocoa.mm

```diff
@@ -166,22 +166,40 @@ id toNSObject(JSContextRef context, JSValueRef valueRef, Class requiredClass)
 
     JSValue *value = [JSValue valueWithJSValueRef:valueRef inContext:[JSContext contextWithJSGlobalContextRef:JSContextGetGlobalContext(context)]];
 
+    if (value.isArray) {
+        if (requiredClass && ![requiredClass isSubclassOfClass:NSArray.class])
```

Comment:
I made a mistake when looking at `requiredClass` when I told you to do it this way. `requiredClass` is really `containingObjectsOfClass`. So this method should be:

```objc++
id toNSObject(JSContextRef context, JSValueRef valueRef, Class containingObjectsOfClass)
{
    ASSERT(context);

    if (!valueRef)
        return nil;

    JSValue *value = [JSValue valueWithJSValueRef:valueRef inContext:[JSContext contextWithJSGlobalContextRef:JSContextGetGlobalContext(context)]];

    if (value.isArray) {
        NSUInteger length = [value[@"length"] toUInt32];
        NSMutableArray *resultArray = [NSMutableArray arrayWithCapacity:length];

        for (NSUInteger i = 0; i < length; ++i) {
            JSValue *itemValue = [value valueAtIndex:i];
            if (id convertedItem = toNSObject(context, itemValue.JSValueRef))
                [resultArray addObject:convertedItem];
        }

        if (!containingObjectsOfClass || containingObjectsOfClass == NSObject.class)
            return [resultArray copy];

        return filterObjects(resultArray, ^bool(id _, id value) {
            return [value isKindOfClass:containingObjectsOfClass];
        });
    }

    if (value._isDictionary)
        return toNSDictionary(context, valueRef);

    if (value.isObject && !value.isDate && !value.isNull)
        return value;

    return [value toObject];
}
```